src/Access/SettingsProfilesCache.cpp

```diff
@@ -184,6 +187,11 @@ void SettingsProfilesCache::substituteProfiles(
         names_of_substituted_profiles.emplace(profile_id, profile->getName());
     }
     std::reverse(substituted_profiles.begin(), substituted_profiles.end());
+
+    boost::range::remove_erase_if(profiles, [&substituted_profiles_set](const UUID & profile_id)
```

Comment:
Can that be done with some STL built-in instead of boost?

Reply:
```
boost::range::remove_erase_if(profiles, func);
```
seems shorter and clearer than
```
profiles.erase(std::remove_if(profiles.begin(), profiles.end(), func), profiles.end());
```

Reply:
Turns out, there is a relatively new (C++20) [`std::erase_if`](https://en.cppreference.com/w/cpp/container/vector/erase2), which is equivalent to the `boost::range::remove_erase_if`